src/components/views/right_panel/UserInfo.tsx

```diff
@@ -121,13 +122,16 @@ export const getE2EStatus = (cli: MatrixClient, userId: string, devices: IDevice
     return anyDeviceUnverified ? E2EStatus.Warning : E2EStatus.Verified;
 };
 
-async function openDMForUser(matrixClient: MatrixClient, user: RoomMember): Promise<void> {
-    const startDMUser = new DirectoryMember({
+/**
+ * Converts the member to a DirectoryMember and starts a DM with them.
+ */
+async function openDmForUser(matrixClient: MatrixClient, user: Member): Promise<void> {
```

Comment:
Could we fix the types on `MessageButton` & `UserOptionsSection` to also use this Member type so that TS can catch these bugs next time

Reply:
If it is OK for you I would do this in a separate PR. This focuses on fixing the release blocker.

Reply:
I'd argue it is part of the validation for the fix. The test is all well and good but fixing the types also confirms that nothing else in this vein was introduced. Avatars aren't the only place User and RoomMember objects differ